docs/v1/api.md

```diff
@@ -0,0 +1,230 @@
+# Service Catalog API
+
+This document proposes a Kubernetes-native service broker API. It is a formalization of [@krancour](http://github.com/krancour)'s [presentation](https://docs.google.com/presentation/d/1Fm3qG9zQ4R8ZbhtTObmQjO3UZYJ0ruYI4qr5w_gEKN8/edit?usp=sharing) that he gave at the sig-service-catalog meeting on 10/10/2016.
```

Comment:
Also, once this is ready for merge we should have another PR that rephrases this as a design document, rather than a proposal.  In the main project we differentiate between the two and move proposals to design when they're implemented.  When we were starting out the SIG that seemd a little heavy to begin with, so there is no 'proposals' directory in this project.  Just a note.